config/initializers/settings_schema.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+Dir.glob('app/workers/*.rb').sort.each do |file|
+  require_relative Rails.root.join(file).to_s
+end
+
+# The Schema class to validate the Settings against
+class SettingsSchema < Dry::Validation::Schema
+  configure do |config|
+    config.messages_file = Rails.root.join(
+      'config/initializers/settings_validation_errors.yml'
+    )
+  end
+
+  def scheme?(list, value)
+    Array(list).include?(URI(value).scheme)
+  end
+
+  def absolute?(value)
+    URI(value).absolute?
+  end
+
+  def no_path?(value)
+    URI(value).path.empty?
+  end
+
+  def no_query?(value)
+    URI(value).query.nil?
+  end
+
+  def no_fragment?(value)
+    URI(value).fragment.nil?
+  end
+
+  def no_userinfo?(value)
+    URI(value).userinfo.nil?
+  end
+
+  def directory?(value)
+    File.directory?(value)
+  end
+
+  def create_directory?(dir)
```

Comment:
I would rather call this `create_directory` without the question mark because there is an action that is performed and not only a check.

Reply:
Unfortunately this does not work with dry-validation. It must be suffixed with `?`.

Reply:
Ah, ok